0001-remove-spandata.md

```diff
@@ -0,0 +1,32 @@
+# Remove SpanData
+
+Remove and replace SpanData by adding span start and finish options.
+ 
+
+## Motivation
+
+SpanData represents an immutable span object, creating a fairly large API for all of the fields (12 to be exact). It exposes what feels like an SDK concern and implementation detail to the API surface. As a user, this is another API I need to learn how to use. As an implementer, it’s a new data type that needs to be supported. The primary motivation for removing SpanData revolves around the desire to reduce the size of the tracing API.
```

Comment:
Note the connection with https://github.com/open-telemetry/opentelemetry-specification/issues/177

In the context of this discussion, the only change here is "rename SpanData to the OT concept of StartSpanOptions". I think this deserves emphasis.

Reply:
I'd be happy to pair up on writing another paragraph on this (tomorrow).